src/execution/perfect_aggregate_hashtable.cpp

```diff
@@ -249,6 +264,15 @@ void PerfectAggregateHashTable::Scan(idx_t &scan_position, DataChunk &result) {
 	// then construct the payloads
 	result.SetCardinality(entry_count);
 	RowOperations::FinalizeStates(layout, addresses, result, grouping_columns);
+
+#ifdef LINEAGE
+  // Log group_values & count for this chunk. This maps output to groups
+  unique_ptr<uint32_t[]> group_values_ptr(new uint32_t[entry_count]);
+  std::copy(group_values, group_values + entry_count, group_values_ptr.get());
```

Comment:
Can we do this as an std::move instead of an std::copy so we avoid needing to allocate && set a full copy? Or does that not work? (I'm still not 100% clear on copy/move dynamics 😅 )

Reply:
hm I am not sure since group_values is statically allocated.